### ISSUE TITLE
`blockcommit` and `blockpull` operations for external snapshots

### ISSUE THREAD

**Issue by kvaps** [OpenNebula/one#1507]
# Component
- [X] Core
- [ ] Authorization (LDAP, x509 certs...)
- [ ] Command Line Interface (CLI)
- [ ] Contextualization
- [ ] Documentation
- [ ] KVM
- [ ] Networking
- [ ] Orchestration (OpenNebula Flow)
- [X] Sunstone
- [ ] Storage
- [ ] Scheduler
- [ ] vCenter

# Enhancement Request
## Description
I want to have opportunity merging external disk snapshots.
It's possible by using `blockcommit` and `blockpull` commands in virsh. It would be nice for have support from opennebula side.
More info here:
* [Efficient live disk backup with active blockcommit](http://wiki.libvirt.org/page/Live-disk-backup-with-active-blockcommit)
* [Merge an entire chain (including current active image) into its base image](http://wiki.libvirt.org/page/Live-merge-an-entire-disk-image-chain-including-current-active-disk)
* [External Snapshot management](http://wiki.libvirt.org/page/Live-merge-an-entire-disk-image-chain-including-current-active-disk)
## Use case
You can create live qcow2 snapshots even for raw images, and painlessly merge it without rebooting.
It's already supported by qemu and libvirt, we should add support for opennebula

**Comment by xprxprx** [comment]
This feature is still interesting to be done.

**Comment by rsmontero** [comment]
With qcow2 we allow any revert operation, so you can end up with a backing file dependency as the one shown:

```
┌──────────┐     ┌──┐     ┌──┐    ┌─────────┐
│Base Image│◄────┤1 │◄────┤3 ◄────┤ WR SNAP │
└──────────┘     └──┘     └──┘    └─────────┘
                   ▲        ▲
                   │        │
                   │        │
                 ┌─┴┐     ┌─┴┐
                 │ 2│     │ 4│
                 └──┘     └──┘
```
(WR SNAP is the file where the VM is actually writing changes)

Let's say I want to remove snapshot 3, I cannot commit to snapshot 1 (i.e. merge contents of 3 into 1) because it has children (2 in this case that will break). So, in general I think we cannot benefit of the block commit operation. However we could  use for chains like this one:
```
┌──────────┐     ┌──┐     ┌──┐    ┌─────────┐
│Base Image│◄────┤1 │◄────┤3 ◄────┤ WR SNAP │
└──────────┘     └──┘     └──┘    └─────────┘
```

We are using the blockcommit operation for the backup interface with qcow2 images, but We don't see its application to the current snapshot structure. (We could apply it if the revert operation drop the children snapshots but that would require a change in the behavior of qcow2 snapshots).

These operations are being used to improve the backup interface of qcow2 based VMs, though.

**Comment by rsmontero** [comment]
Finally we have implemented this feature as suggested details in #6265
